service/host.go

```diff
@@ -201,11 +201,20 @@ func (s *host) AddModules(modules ...ModuleCreateFunc) error {
 	return nil
 }
 
-// Start runs the serve loop. If Shutdown() was called then the shutdown reason
-// will be returned.
-func (s *host) Start(waitForShutdown bool) (<-chan Exit, <-chan struct{}, error) {
-	// TODO(glib): this function has a lot of different exits and different
-	// mutex unlock spots, maybe a rewrite is in order?
+// StartAsync service is used as a non-blocking the call on service start. StartAsync will
+// return the call to the caller with a Control to listen on channels
+// and trigger manual shutdowns.
+func (s *host) StartAsync() Control {
+	return s.start(false)
+}
+
+// Start service is used for blocking the call on service start. Start will block the
+// call and yield the control to the service lifecyce manager.
+func (s *host) Start() {
+	s.start(true)
+}
+
+func (s *host) start(waitForShutdown bool) Control {
```

Comment:
since we are splitting control into async and blocking, i think all the code that relates to blocking should be removed from this method and places into `Start()`. This will make the method a little less cluttered